python/pyspark/worker.py

```diff
@@ -22,7 +22,12 @@
 import os
 import sys
 import time
-import resource
+# 'resource' is a Unix specific module.
+has_resource_module = True
+try:
+    import resource
+except ImportError:
+    has_resource_module = False
```

Comment:
@vanzin, I'm going to remove this flag. Does it then look okay to you?

Reply:
I'm -1 on this change.

I think the correct behavior is that Python should not fail if resource cannot be imported, and the JVM should not do anything differently.

Reply:
I'm -1 too. Remove the check from the Java side. Let Python deal with whether it can enforce resource limits or not. The JVM should just tell Python what the limit should be. I don't understand why is that so controversial.

Reply:
-1 for adding only Python check from my side. I don't understand why it's controversial as well. Python can't fail because `resource` only exists in unix based systems. We only have one exception case Windows which we know in JVM side.

Configuration control should better be done in JVM side if possible and I don't think we should allow this feature work differently on Windows and this should be tested before we document we support it.

Reply:
For instance, forking in Python is disallowed in Python so we do that control in JVM side.

Reply:
What is the behavior you want? What do you mean "python can't fail"?

If the memory limit is set by the user, you have two options when the python side cannot do anything with it:

- ignore it (the current patch)
- raise an error

Both can be done from the python side and do not require any checks in the JVM.

Checking in the JVM is the wrong thing. You're hardcoding the logic that this feature can never work on Windows. And if someone finds out that it can, then you'll have to change it, and that check was basically useless, since an equivalent check exists on the python side.

Reply:
Can I add some more people for input? Let me just follow majority. cc @ueshin, @BryanCutler.

Reply:
> For instance, forking in Python is disallowed in Python on Windows

That is *completely* different. Because that affects how the Java side talks to the Python side.

The feature in this PR has no bearing on what happens on the JVM side at all.

Reply:
> Let me just follow majority

A -1 is a -1. It's not majority based (this is not a release).

Reply:
What I meant by "What do you mean "python can't fail"? is quoted from `Python should not fail if resource cannot be imported` from @rdblue.

I left the last comment before seeing your comment @vanzin. Let me leave an answer for that soon.

Reply:
Yea, so the check and be done in Python side and JVM side. Yes, I am explicitly disabling it on Windows. I think the proper assumption is that we note it doesn't work if that's not tested. If someone finds it working, then that patch explicitly enables it with a proper documentation.

Reply:
I really don't understand what you said above. If the check is done on the Python side, what is the exact reason why you need it on the JVM side?

Reply:
> That is completely different. Because that affects how the Java side talks to the Python side.

Not completely different tho. Ideally we can have one script that handles both cases. It deals with OS specific differences in Python sides.

Reply:
Just for consistency, and to avoid that value is set in JVM but works differently in Python side.

Reply:
I think it's way better to have the JVM not care and have the python side handle the setting according to its capabilities.

It's a smaller change, it's more localized, and it reduces the amount of changes if this needs to be tweaked again in the future.

I don't see the consistency argument you're making. You picked one example that is very different from this one, since it actually affects the behavior of the JVM code. Could it have been developed differently? Of course. But we're not discussing that feature here.

And there are so few of these cases that you really can't draw a pattern from them.

Reply:
OK. let me remove that check here for now. I think we can discuss about the whole problem about how it's implemented later.